cmd/veneur/main.go

```diff
@@ -120,9 +117,7 @@ func main() {
 			}).Error("Error reading from UDP")
 			continue
 		}
-		select {
-		case parserChan <- buf[:n]:
-		}
+		parserChan <- buf[:n] // TODO: termination condition for this channel?
```

Comment:
oh, you wanted those to be nonblocking? you need a default case for that:

```go
select {
   case parserChan <- buf[:n]:
   default:
}
```

 A select with only one case is equivalent to the case itself. Note that a nonblocking send will just drop, ie if it's not ready to send then it will never send at all. You usually use this in combination with buffered channels (ie send until the capacity is exceeded and then just drop messages). I think you definitely wanted a hard blocking send here.